### DISCUSSION TITLE
Set Seed for Random MPS

### DISCUSSION THREAD

**Discussion by jtcantin** [block-hczhai/block2-preview#79] [Q&A] [answered]
Hello,

Is there a way to set the seed for the random number generator used in `DMRGDriver.get_random_mps()`?

There doesn't seem to be a function argument related to the seed and nothing appeared when I searched for "seed" in the docs.

Thank you!

**Comment by hczhai** [comment, marked as answer]
You can use ``driver.bw.b.Random.rand_seed`` to set a random seed before calling ``driver.get_random_mps``. For example:

```python3
from pyblock2.driver.core import DMRGDriver, SymmetryTypes
from pyblock2.algebra.io import MPSTools

ncas, n_elec, spin = 10, 10, 0
driver = DMRGDriver(scratch="./tmp", symm_type=SymmetryTypes.SZ, n_threads=4)
driver.initialize_system(n_sites=ncas, n_elec=n_elec, spin=spin)

driver.bw.b.Random.rand_seed(1234)
ket = driver.get_random_mps(tag="GS", bond_dim=10, nroots=1, dot=1, occs=[0.5] * ncas)
pyket = MPSTools.from_block2(ket)
print('random block at site 1 =\n', pyket.tensors[1].blocks[0].reduced)
```

Note:

1. ``driver.bw.b.Random.rand_seed(0)`` will generate different results every time. Non-zero parameters (such as 1234) will be used as the random seed and you will get the same random MPS when the random seed is the same.
2. Even when the initial MPS can be fixed this way, there is no guarantee that the optimized MPS after performing DMRG is always the same. This is due to the multi-threading (the order of floating-point arithmetic can depend on the dynamic thread scheduling).
